indra/explanation/pathfinding/util.py

```diff
@@ -113,3 +115,47 @@ def get_sorted_neighbors(G, node, reverse, force_edges=None):
             key=lambda n:
                 G.edges[(node, n)].get('belief', 0),
             reverse=True)
+
+
+edge_filter_functions = {}
+# Need to set a global G variable for filter edge function to work
+G = None
+
+
+def register_edge_filter(function):
+    """
+    Decorator to register a function as an edge filter for getting subgraphs.
+    Edge filter function should take two (u, v - for DiGraph) or three
+    (u, v, k -  for MultiDiGraph) parameters and return True if the edge
+    should be in the graph and False otherwise.
+    """
+    edge_filter_functions[function.__name__] = function
+    return function
+
+
+def get_subgraph(g, filter_func_name):
+    """Get a subgraph of original graph filtered by a provided function."""
+    # Updating global variable here to handle NetworkX implementation of
+    # closures used in subgraph_view
+    global G
+    G = g
+    filter_edge = edge_filter_functions.get(filter_func_name)
+    view = nx.subgraph_view(g, filter_edge=filter_edge)
```

Comment:
ben's comment was it would be better to do something like:

```suggestion
    view = nx.subgraph_view(g, filter_edge=functools.partial(filter_edge, g))
```

And change the interface of `filter_edge` to also have an argument for the graph in the first position.